### ISSUE TITLE
Error: "no objects visited" while installing the chart using helm3 command.

### ISSUE THREAD

**Issue by ekumari** [helm/helm#7632]
Output of `helm version`: version.BuildInfo{Version:"v3.1.0-rc.1", GitCommit:"0977ded29a00383db5cbfd91ae220d30457f6ddb", GitTreeState:"clean", GoVersion:"go1.13.7"}

Output of `kubectl version`: 
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-21T22:17:28Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.8-gke.33", GitCommit:"2c6d0ee462cee7609113bf9e175c107599d5213f", GitTreeState:"clean", BuildDate:"2020-01-15T17:47:46Z", GoVersion:"go1.12.11b4", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  GKE

**Comment by hickeyma** [comment]
@ekumari Could you provide more details on what you are trying to install, the com mand used and the output of the command?

**Comment by suciuandrei94** [comment]
I am having exactly the same problem.

I tried to do the following things:

`image: "{{ required "A valid baseRepository is required!" .Values.image.baseRepository }}/{{.Values.image.name}}:{{ .Chart.AppVersion }}"`

`{{- if .Values.persistence }}
{{- if (ne .Values.persistence.disk "0") }}`

Each of these statements, although correct (according to helm docs) result in `Error: "no objects visited"`

I've tried installing the chart without those (by commenting them) and then did a helm upgrade afterwards (with these uncommented) and then it just suddenly removed my deployment + pod (there is no failure reported).

I could help reproducing if needed but it is pretty simple.

**Comment by srisurya777** [comment]
@ekumari there should be some error in your templates. Is the 'helm lint' successful with zero failures ?

**Comment by ekumari** [comment]
@srisurya777  Yes helm lint is returning error in chart.yaml file

**Comment by ekumari** [comment]
@hickeyma  I ran "helm install  testinfo informix/ --namespace jam --set global.sofyImagePullSecret=sofypullsecret --set global.FLEXNET_CLS_URL=LICENSE_URL --set global.FLEXNET_CLS_ID=SOME_ID" using helm3 version

Output : no objects visited

**Comment by brandonferrer** [comment]
I'm having the same problem, but my chart passes lint with no failures. 

Helm version:
version.BuildInfo{Version:"v3.1.0", GitCommit:"b29d20baf09943e134c2fa5e1e1cab3bf93315fa", GitTreeState:"clean", GoVersion:"go1.13.8"}

kubectl version:
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:40:16Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}

**Comment by hickeyma** [comment]
@ekumari If you use `--sry-run --debug` flags, does it provide any more information to find where the issue is? What is the output when you run it like this?

**Comment by suciuandrei94** [comment]
I reckon he gets same behaviour as me:

```
andrei@andrei-Precision-5540:~/template-boot-service$ helm upgrade --install -f helm/env/dev/values.yaml template-boot-service helm/template-boot-service/ --dry-run
Release "template-boot-service" does not exist. Installing it now.
NAME: template-boot-service
LAST DEPLOYED: Thu Feb 20 10:13:55 2020
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
HOOKS:
MANIFEST:

andrei@andrei-Precision-5540:~/Documents/projects/template-boot-service$ helm upgrade --install -f helm/env/dev/values.yaml emplate-boot-service helm/emplate-boot-service/
Release "template-boot-service" does not exist. Installing it now.
Error: no objects visited
```

So no error thrown during dry run

**Comment by brandonferrer** [comment]
> @srisurya777 Yes helm lint is returning error in chart.yaml file

What is the error? 

I had a similar issue, but my chart passed lint. I was getting Error: no objects visited because my chart was missing a required value. Something to try/rule out. The error isn't helpful at all, but I was able to figure it out by the lint output. 

```
helm lint datadog-cluster
engine.go:159: [INFO] Missing required value: Required: apiKey
engine.go:159: [INFO] Missing required value: Required: appKey
engine.go:159: [INFO] Missing required value: Required: dcaSecret
==> Linting datadog-cluster
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
```

**Comment by suciuandrei94** [comment]
@brandonferrer is correct, with helm lint it gives right output but helm upgrade doesn't give the reason for failure.

```
andrei@andrei-Precision-5540:~/Documents/projects/template-boot-service$ helm lint helm/template-boot-service/
engine.go:159: [INFO] Missing required value: A valid baseRepository is required!
```

**Comment by brandonferrer** [comment]
@suciuandrei94 just a thought, make sure baseRepository is defined in your values.yaml or you are passing it in your install/upgrade command --set baseRepository=<VALUE>

**Comment by suciuandrei94** [comment]
Yes, but I did not pass it on purpose, I want the dev ops guys to know that they missed to set the baseRepository, however instead of relevant error I get "no objects visisted" :)

**Comment by hickeyma** [comment]
@brandonferrer @suciuandrei94 The error might not be showing due to #7596. Do you mind testing with the latest in master and see of the error gets shown?

**Comment by suciuandrei94** [comment]
Tried it out @hickeyma 
```andrei@andrei-Precision-5540:~/work/src/github.com/helm/helm$ ./bin/helm upgrade --install -f /home/andrei/Documents/projects/template-boot-service/helm/env/dev/values.yaml template-boot-service /home/andrei/Documents/projects/template-boot-service/helm/template-boot-service/
Release "emplate-boot-service" does not exist. Installing it now.
Error: execution error at (template-boot-service/templates/deployment.yaml:50:21): A valid baseRepository is required!
```

So it works fine with master version.

**Comment by hickeyma** [comment]
That's good to hear, thanks @suciuandrei94

**Comment by bacongobbler** [comment]
Fixed in 3.1.1
